library/core/src/num/int_macros.rs

```diff
@@ -1167,6 +1167,7 @@ macro_rules! int_impl {
         #[rustc_const_unstable(feature = "unchecked_neg", issue = "85122")]
         #[inline(always)]
         #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+        #[requires(self != $SelfT::MIN)]
```

Comment:
Can we also add an `ensures` clause that checks that the result is what we expect it to be? Even though the focus in this challenge is on memory safety, but the `ensures` clause in this case should be simple enough to add as part of this PR.

Reply:
Of course. Does `ensures(self == -self)` make sense in this case?

Reply:
An `ensures` clause is in terms of a `result`, so I think the following should work:
```rust
#[ensures(|result| result == -self)]
```

Reply:
Oops, yes. I did mean that, in terms of the result. Thanks for clarifying! I'll make the change now.

Reply:
Thanks. Good to go.